packages/language-server/src/plugins/typescript/features/FindComponentUsagesProvider.ts

```diff
@@ -1,105 +1,79 @@
 import { Location, Position, Range } from 'vscode-languageserver';
 import { URI } from 'vscode-uri';
-import { pathToUrl } from '../../../utils';
+import { flatten, pathToUrl } from '../../../utils';
 import { FindComponentUsagesProvider } from '../../interfaces';
 import { LSAndTSDocResolver } from '../LSAndTSDocResolver';
-import {
-    convertToLocationRange,
-    convertToTextSpan,
-    hasNonZeroRange,
-    isSvelteFilePath
-} from '../utils';
-import { FindReferencesProviderImpl } from './FindReferencesProvider';
-import { SnapshotFragmentMap } from './utils';
+import { convertToLocationRange, hasNonZeroRange } from '../utils';
 import { Document, DocumentManager } from '../../../lib/documents';
-import ts, { ReferenceEntry, TextSpan } from 'typescript';
-import { SnapshotFragment } from '../DocumentSnapshot';
-
+import ts from 'typescript';
+import { isNoTextSpanInGeneratedCode, SnapshotFragmentMap } from './utils';
+import { SvelteSnapshotFragment } from '../DocumentSnapshot';
+import { lsConfig } from '../../../ls-config';
 export class FindComponentUsagesProviderImpl implements FindComponentUsagesProvider {
     constructor(private readonly lsAndTsDocResolver: LSAndTSDocResolver) {}
 
     async findComponentUsages(uri: string): Promise<Location[] | null> {
         const u = URI.parse(uri);
         const fileName = u.fsPath;
-        const lang = await this.getLSForPath(fileName);
+        const document = await this.getDocument(fileName);
+        const { lang, tsDoc } = await this.getLSAndTSDoc(document);
+        const fragment = tsDoc.getFragment();
+        const ignoreImports =
```

Comment:
Is this the right way to pull user configuration? It didn't seem like it was switching on/off when I changed my vscode setting for it